src/components/ExportDialog/ExportDialog.js

```diff
@@ -24,32 +24,35 @@ import ExportTargetsSelector from './ExportTargetsSelector'
 import { propTypes, defaultProps } from './ExportDialog.props'
 
 class ExportDialog extends PureComponent {
-  handlers = {}
+  constructor(props) {
+    super()
 
-  state = {
-    currentTargets: [],
+    const { type } = props
+    this.state = {
+      currentTargets: [type],
+    }
   }
 
-  onTargetSelect = (target) => {
-    if (!this.handlers[target]) {
-      this.handlers[target] = () => {
-        const { currentTargets } = this.state
-        if (!currentTargets.includes(target)) {
-          this.setState({ currentTargets: [...currentTargets, target] })
-        }
-      }
+  componentWillReceiveProps(nextProps) {
```

Comment:
componentWillReceiveProps is deprecated :/

Reply:
just an idea how to change to getDerivedStateFromProps ( didn't try to run 😆 )
```js
getDerivedStateFromProps(nextProps, prevState) {
    if(nextProps.type !== prevState.type) {
      return  { currentTargets: [nextProps.type] }
   }
   return { type: nextProps.type }
}
```

Reply:
thanks! always forgetting `componentWillReceiveProps` will be gone soon, such a useful method... :(

Reply:
yeah and getDerivedStateFromProps need a bit of hack to get prevProps LOL